tasker/executors/parallel_executor.py

```
@@ -370,16 +386,34 @@ def execute_parallel_tasks(parallel_task, executor_instance):
                                 task_display_id = f"{task_id}-{result['task_id']}"
-                                executor_instance.log(f"Task {task_display_id}: Sleeping for {sleep_seconds} seconds...")
-                                time.sleep(sleep_seconds)
-                            
+                                executor_instance.log(f"Task {task_display_id}: Scheduling non-blocking sleep for {sleep_seconds} seconds...")
+
+                                # Use non-blocking sleep with deferred result processing
+                                result_completion_event = threading.Event()
+
+                                def sleep_completed_callback():
+                                    """Process result after sleep completes."""
+                                    executor_instance.log_debug(f"Task {task_display_id}: Sleep completed, processing result")
```

Ruff (0.13.2)

Function definition does not bind loop variable `task_display_id`

(B023)